### PR TITLE
GEN-1163 - feat: added TrialExtensionForm

### PR DESCRIPTION
**Pull request by guilhermespopolin**
## Describe your changes

* Added `TrialExtensionForm` component - right now it only covers the "dismissed extension" state.

Screenshot 2023-09-20 at 15 45 25

## Justify why they are needed

Part of Car dealership project.

### CODE REVIEW

**Comment by guilhermespopolin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #3249** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3249" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3249?utm_source=stack-comment).

File: apps/store/src/features/carDealership/TrialExtensionForm.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import { SpaceFlex } from '@/components/SpaceFlex/SpaceFlex'
+
+type Props = { onUndo: () => void } & Pick<
+  ComponentProps<typeof ProductItemContractContainerCar>,
+  'contract'
+>
+
+export const TrialExtensionForm = (props: Props) => {
+  const handleUndo = () => {
+    datadogRum.addAction('Car dealership | ndo extension removal')
```
Comment (robinandeer) [inline apps/store/src/features/carDealership/TrialExtensionForm.tsx:14]: nit: spelling mistake "ndo"

File: apps/store/src/features/carDealership/TrialExtensionForm.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import { datadogRum } from '@datadog/browser-rum'
+import { type ComponentProps } from 'react'
+import { Space, Button, RestartIcon } from 'ui'
+import { ProductItemContractContainerCar } from '@/components/ProductItem/ProductItemContractContainer'
```
Comment (robinandeer) [inline apps/store/src/features/carDealership/TrialExtensionForm.tsx:4]: nit: let's move this inside the feature folder

**Review by robinandeer — APPROVED** [review]
looks good, let's keep building on this!

**Comment by guilhermespopolin** [comment]
### Merge Activity

* **Sep 20, 11:17 AM**: @guilhermespopolin started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3249).

**Pull request merged**